File: app/src/main/java/com/example/testtanaw/util/SDGAdapter2.java
package com.example.testtanaw.util;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import androidx.recyclerview.widget.RecyclerView;
import com.example.testtanaw.R;
import java.util.List;

public class SDGAdapter2 extends RecyclerView.Adapter<SDGAdapter2.SDGViewHolder> {
    private final List<String> sdgImages;

    public SDGAdapter2(List<String> sdgImages) {
        this.sdgImages = sdgImages;
    }

    public class SDGViewHolder extends RecyclerView.ViewHolder {
        public final ImageView imageView;

        public SDGViewHolder(View view) {
            super(view);
            imageView = view.findViewById(R.id.sdgImageView);
        }
    }

    @Override
    public SDGViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_sdg_circle, parent, false);
        return new SDGViewHolder(view);
    }

    @Override
    public void onBindViewHolder(SDGViewHolder holder, int position) {
        int resourceId = holder.itemView.getContext().getResources().getIdentifier(
            sdgImages.get(position), "drawable", holder.itemView.getContext().getPackageName()
        );
        holder.imageView.setImageResource(resourceId);
    }

    @Override
    public int getItemCount() {
        return sdgImages.size();
    }
}

File: app/src/main/java/com/example/testtanaw/SDGAdapter2.kt
package com.example.testtanaw

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView

class SDGAdapter2(private val sdgImages: List<String>) :
    RecyclerView.Adapter<SDGAdapter2.SDGViewHolder>() {

    inner class SDGViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val imageView: ImageView = view.findViewById(R.id.sdgImageView)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SDGViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_sdg_circle, parent, false)
        return SDGViewHolder(view)
    }

    override fun onBindViewHolder(holder: SDGViewHolder, position: Int) {
        val resourceId = holder.itemView.context.resources.getIdentifier(
            sdgImages[position], "drawable", holder.itemView.context.packageName
        )
        holder.imageView.setImageResource(resourceId)
    }

    override fun getItemCount() = sdgImages.size
}
